File: sol/solution/0001-0100/0038/Solution.scala
object Solution {
    def countAndSay(n: Int): String = {
        var res = "1"
        for (_ <- 2 to n) {
            val sb = new StringBuilder
            var cnt = 1
            for (j <- 1 until res.length) {
                if (res(j) == res(j-1)) cnt += 1
                else {
                    sb.append(cnt).append(res(j-1))
                    cnt = 1
                }
            }
            sb.append(cnt).append(res.last)
            res = sb.toString
        }
        res
    }
}

File: sol/solution/0001-0100/0038/Solution.cpp
class Solution {
public:
    string countAndSay(int n) {
        string res = "1";
        for (int i = 2; i <= n; i++) {
            string cur = "";
            int cnt = 1;
            for (int j = 1; j < res.size(); j++) {
                if (res[j] == res[j-1]) {
                    cnt++;
                } else {
                    cur += to_string(cnt) + res[j-1];
                    cnt = 1;
                }
            }
            cur += to_string(cnt) + res.back();
            res = cur;
        }
        return res;
    }
};